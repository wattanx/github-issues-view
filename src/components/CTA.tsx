import NextLink from 'next/link';
import { FaArrowRight } from 'react-icons/fa';
import { DiGithubBadge } from 'react-icons/di';
import { Container, Button, Stack } from 'components';

export const CTA = () => (
  <Container
    flexDirection={['unset', 'row', 'row', 'row']}
    width="100%"
    maxWidth="48rem"
    py={3}
  >
    <Stack
      mt="10"
      spacing="4"
      justify="center"
      direction={{ base: 'column', sm: 'row' }}
    >
      <NextLink href="/issues" passHref>
        <Button
          h="4rem"
          px="40px"
          fontSize="1.2rem"
          as="a"
          size="lg"
          bg="teal.200"
          rightIcon={<FaArrowRight fontSize="0.8em" />}
        >
          Get Started
        </Button>
      </NextLink>

      <Button
        as="a"
        size="lg"
        h="4rem"
        px="40px"
        fontSize="1.2rem"
        href="https://github.com/wattanx/github-issues-view"
        target="__blank"
        leftIcon={<DiGithubBadge size="1.5em" />}
      >
        GitHub
      </Button>
    </Stack>
  </Container>
);
