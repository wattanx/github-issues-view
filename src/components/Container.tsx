import { Box, BoxProps } from 'components';

export const Container = (props: BoxProps) => {
  return (
    <Box
      w="full"
      pb="12"
      pt="3"
      maxW={{ base: 'xl', md: '7xl' }}
      mx="auto"
      px={{ base: '6', md: '8' }}
      {...props}
    />
  );
};
