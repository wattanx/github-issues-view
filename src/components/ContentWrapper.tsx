import { Box } from '../components';

export type ContentWrapperProps = {
  children: React.ReactNode;
};
export const ContentWrapper: React.FC<ContentWrapperProps> = ({ children }) => (
  <Box maxWidth="1280px" margin="auto" padding={5}>
    {children}
  </Box>
);
