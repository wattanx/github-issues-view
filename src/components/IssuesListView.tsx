import NextLink from 'next/link';
import {
  Box,
  ClosedStatus,
  Flex,
  OpenedStatus,
  Stack,
  Text,
} from '../components';
import { IssueStateType } from 'types/github';
import { IssueType } from 'framework';

export const IssuesListView: React.VFC<{ issues?: IssueType[] }> = ({
  issues,
}) => {
  return (
    <Stack spacing="0">
      {issues &&
        issues.map((x) => {
          return (
            <IssuesListCell
              key={x.issueNumber}
              title={x.title}
              issueNumber={x.issueNumber}
              state={x.state}
            />
          );
        })}
    </Stack>
  );
};

type IssuesListCellProps = {
  state: IssueStateType;
  title: string;
  issueNumber: number;
};

const IssuesListCell: React.VFC<IssuesListCellProps> = ({
  state,
  title,
  issueNumber,
}) => (
  <Flex
    borderTopColor="#e2e2e2"
    borderTopWidth="1px"
    _first={{ borderTopColor: 'transparent' }}
  >
    <Box paddingTop={3} paddingLeft={3}>
      {state === 'open' && <OpenedStatus />}
      {state === 'closed' && <ClosedStatus />}
    </Box>
    <Box padding={2}>
      <NextLink
        href="/issues/[issueNumber]"
        as={`/issues/${issueNumber}`}
        shallow={true}
        passHref
      >
        <Box as="a">
          <Text>{title}</Text>
        </Box>
      </NextLink>
      <Text fontSize="sm">#{issueNumber}</Text>
    </Box>
  </Flex>
);
