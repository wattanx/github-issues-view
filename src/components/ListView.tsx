import { Box } from 'components';

export type ListViewProps = {
  children: React.ReactNode;
};

export type ListViewCellProps = {
  children: React.ReactNode;
};

export const ListView: React.FC<ListViewProps> = ({ children }) => (
  <Box>{children}</Box>
);

export const ListViewCell: React.FC<ListViewCellProps> = ({ children }) => (
  <Box>{children}</Box>
);
