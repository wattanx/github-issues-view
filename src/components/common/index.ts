export * from './icon';
export * from './layout';
export * from './typography';
