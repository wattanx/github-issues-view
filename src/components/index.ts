export * from './common';
export * from './Container';
export * from './ContentWrapper';
export * from './Status';
export * from './IssuesListView';
