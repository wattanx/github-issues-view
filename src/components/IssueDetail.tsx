import { Text, Heading, HStack } from '@chakra-ui/react';
import { Box } from './common';

export type IssueDetailProps = {
  title: string;
  body: string;
  state: string;
  issueNumber: string;
};

export const IssueDetail: React.VFC<IssueDetailProps> = ({
  title,
  body,
  state,
  issueNumber,
}) => {
  return (
    <Box>
      <Box borderBottomWidth="1px" borderBottomColor="#e2e2e2">
        <HStack>
          <Heading>
            {title}
            <Box fontWeight="normal" color="#b9b9b9" as="span" marginLeft={2}>
              #{issueNumber}
            </Box>
          </Heading>
        </HStack>
      </Box>
      <Box rounded="md" border="1px #e2e2e2 solid" marginTop={3} padding={2}>
        <Box borderBottom="1px #e2e2e2 solid" padding={1}></Box>
        <Box marginTop={3} width={['100%', '100%', '100%', '70%']}>
          {body}
        </Box>
      </Box>
    </Box>
  );
};
