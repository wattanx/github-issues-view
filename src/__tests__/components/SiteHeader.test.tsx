import { SiteHeader } from 'components';
import { render } from '@testing-library/react';

test('SiteHeader Snapshot Test', () => {
  const component = render(<SiteHeader />);
  expect(component).toMatchSnapshot();
});
