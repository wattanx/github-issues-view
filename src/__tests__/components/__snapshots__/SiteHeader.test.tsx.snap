// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <header
        class="css-15b2on2"
      >
        <div
          class="css-1k68o7q"
        >
          <div
            class="chakra-stack css-1pmaxwi"
          >
            <div
              class="css-1lzgxcb"
            >
              <a
                class="css-1z0hpko"
                href="/"
              >
                GitHub Issues View
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="css-15b2on2"
    >
      <div
        class="css-1k68o7q"
      >
        <div
          class="chakra-stack css-1pmaxwi"
        >
          <div
            class="css-1lzgxcb"
          >
            <a
              class="css-1z0hpko"
              href="/"
            >
              GitHub Issues View
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
